crates/vision/src/robot_detection.rs

```diff
@@ -1,332 +1,236 @@
         })
         .collect()
 }
 
-fn cluster_scored_cluster_points(
-    cluster_points: Vec<ScoredClusterPoint>,
-    maximum_cluster_distance: f32,
-    cluster_distance_score_range: &Range<f32>,
-    amount_score_exponent: f32,
-    luminance_score_exponent: f32,
-) -> Vec<Vec<ScoredClusterPoint>> {
-    let distance_score_offset = cluster_distance_score_range.start;
-    let distance_score_scale =
-        cluster_distance_score_range.end - cluster_distance_score_range.start;
-
-    let mut clusters: Vec<Vec<ScoredClusterPoint>> = vec![];
-    for point in cluster_points {
-        let nearest_cluster = clusters
-            .iter_mut()
-            .map(|cluster_points| {
-                let adjusted_distance = cluster_points
-                    .iter()
-                    .map(|cluster_point| {
-                        let score = (point.amount_score.powf(amount_score_exponent)
-                            * point.luminance_score.powf(luminance_score_exponent))
-                            * distance_score_scale
-                            + distance_score_offset;
-                        distance(&cluster_point.point, &point.point) / score
-                    })
-                    .min_by(|left, right| left.total_cmp(right))
-                    .expect("Unexpected empty cluster");
-                (cluster_points, adjusted_distance)
-            })
-            .filter(|(_cluster, adjusted_distance)| *adjusted_distance <= maximum_cluster_distance)
-            .min_by(
-                |(_left_cluster, left_adjusted_distance),
-                 (_right_cluster, right_adjusted_distance)| {
-                    left_adjusted_distance.total_cmp(right_adjusted_distance)
-                },
-            );
-        match nearest_cluster {
-            Some((nearest_cluster, _adjusted_distance_to_nearest_cluster)) => {
-                nearest_cluster.push(point)
-            }
-            None => clusters.push(vec![point]),
-        }
-    }
-    clusters
-}
-
-fn map_clustered_cluster_points_to_scored_clusters(
-    clustered_cluster_points: Vec<Vec<ScoredClusterPoint>>,
-) -> Vec<ScoredCluster> {
-    clustered_cluster_points
-        .into_iter()
-        .map(|cluster_points| {
-            let (mut xs, mut ys): (Vec<_>, Vec<_>) = cluster_points
-                .iter()
-                .map(|point| (point.point.x, point.point.y))
-                .unzip();
-            xs.sort_by(|left, right| left.total_cmp(right));
-            ys.sort_by(|left, right| left.total_cmp(right));
-            let median = point![xs[xs.len() / 2], ys[ys.len() / 2]];
-            let score = cluster_points
-                .iter()
-                .map(|point| point.amount_score * point.luminance_score)
-                .sum();
-            ScoredCluster {
-                center: median,
-                score,
-            }
-        })
-        .collect()
+fn map_to_zero_one(x: f32) -> f32 {
```

Comment:
`map_to_zero_one` sounds IMO like a function with binary output. What do you think about `logistic` or `standard_logistic`?

Reply:
good point, i renamed it